blueflood-core/src/test/java/com/rackspacecloud/blueflood/inputs/processors/TokenDiscoveryWriterTest.java

```diff
@@ -0,0 +1,174 @@
+package com.rackspacecloud.blueflood.inputs.processors;
+
+
+import com.rackspacecloud.blueflood.cache.LocatorCache;
+import com.rackspacecloud.blueflood.cache.TokenCache;
+import com.rackspacecloud.blueflood.concurrent.ThreadPoolBuilder;
+import com.rackspacecloud.blueflood.io.TokenDiscoveryIO;
+import com.rackspacecloud.blueflood.types.IMetric;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.types.Metric;
+import com.rackspacecloud.blueflood.types.Token;
+import com.rackspacecloud.blueflood.utils.TimeValue;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+public class TokenDiscoveryWriterTest {
+
+    @Before
+    public void setUp() throws Exception {
+        LocatorCache.getInstance().resetCache();
+        TokenCache.getInstance().resetCache();
+    }
+
+    @Test
+    public void testGenerateAndConsolidateTokens() {
+
+        String tenantID = "111111";
+        String metricName = "a.b.c.d";
+        Locator locator = Locator.createLocatorFromPathComponents(tenantID, metricName);
+
+        List<List<IMetric>> batch = new ArrayList<>();
+        batch.add(Arrays.asList(new Metric(locator, 1, 1L, new TimeValue(5, TimeUnit.SECONDS), "")));
+
+        Set<Token> tokens = TokenDiscoveryWriter.generateAndConsolidateTokens(TokenDiscoveryWriter.getLocators(batch));
+
+        assertEquals("Invalid number of tokens", locator.getMetricName().split("\\.").length, tokens.size());
+
+    }
+
+    @Test
+    public void testGenerateAndConsolidateTokensMultipleMetrics() {
+
+        Locator locator1 = Locator.createLocatorFromPathComponents("111111", "a.b.c.d");
+        Locator locator2 = Locator.createLocatorFromPathComponents("111111", "a.b.c.e");
+
+        List<List<IMetric>> batch = new ArrayList<>();
+        batch.add(Arrays.asList(new Metric(locator1, 1, 1L, new TimeValue(5, TimeUnit.SECONDS), "")));
+        batch.add(Arrays.asList(new Metric(locator2, 1, 1L, new TimeValue(5, TimeUnit.SECONDS), "")));
+
+        Set<Token> tokens = TokenDiscoveryWriter.generateAndConsolidateTokens(TokenDiscoveryWriter.getLocators(batch));
+
+        assertEquals("Invalid number of tokens", 5, tokens.size());
+
+    }
+
+    @Test
+    public void testProcessTokens() throws Exception {
```

Comment:
More meaningful method name pls? What are we testing here? Either rename method to clarify or add comments.